hnn_core/network_builder.py

```diff
@@ -377,15 +377,15 @@ def _gid_assign(self, rank=None, n_hosts=None):
                 # assigned to this rank
                 for src_gid in self.net.gid_ranges[drive['name']]:
                     conn_idxs = pick_connection(self.net, src_gids=src_gid)
-                    target_gids = list()
+                    target_gids = set()
                     for conn_idx in conn_idxs:
                         gid_pairs = self.net.connectivity[
                             conn_idx]['gid_pairs']
                         if src_gid in gid_pairs:
-                            target_gids += (self.net.connectivity[conn_idx]
-                                            ['gid_pairs'][src_gid])
+                            target_gids.update(self.net.connectivity[conn_idx]
+                                               ['gid_pairs'][src_gid])
 
-                    for target_gid in set(target_gids):
+                    for target_gid in target_gids:
```

Comment:
would suggest using the same idiom everywhere:

```py
list(sorted(set(...)))
```

Reply:
you could even make this a small function in `utils.py` to keep it clean:

```py
def unique_list(x):
    return list(sorted(set(x)))
```

Reply:
This one doesn't actually need be sorted and can stay of type `set`.

Reply:
> This one doesn't actually need to be sorted and can stay of type `set`.

Reply:
For context, the only reason I modified this was because the `target_gids` variable was first instantiated as a list and then used only once here, where it was converted to a set. This is unnecessary and confusing, so I just changed `target_gids` to always be a set.